src/config.py

```diff
@@ -0,0 +1,387 @@
+import os
+import json
+import base64
+
+
+class Config:
+    """Reads Platform.sh configuration from environment variables.
+
+    https://docs.platform.sh/development/variables.html
+
+    The following are 'magic' properties that may exist on a Config object.
+    Before accessing a property, check its existence with hasattr(config, variableName).
+    Attempting to access a nonexistent variable will throw an exception.
+
+    These properties are available at build time and run time:
+
+    project
+        The project ID.
+    applicationName
+        The name of the application, as defined in its configuration.
+    treeId
+        An ID identifying the application tree before it was built: a unique hash
+        is generated based on the contents of the application's files in the repository.
+    appDir
+        The absolute path to the application.
+    entropy
+        A random string generated for each project, useful for generating hash keys.
+
+    These properties are only available at runtime:
+
+    branch
+        The Git branch name.
+    environment
+        The environment ID (usually the Git branch plus a hash).
+    documentRoot
+        The absolute path to the web root of the applicatino.
+    smtpHost
+        The hostname of the Platform.sh defauilt SMTP server (an empty string
+        if emails are disabled on the environment.
+
+    """
+
+    # Local index of the variables that can be accessed as direct properties (build and runtime).
+
+    # The key is the property that will be read. The value is the environment variables, minus
+    # prefix, that contains the value to look up.
+
+    directVariables = {'project': 'PROJECT', 'appDir': 'APP_DIR', 'applicationName': 'APPLICATION_NAME',
```

Comment:
From a readability perspective putting one pair per line is easier to scan visually.  Let's do that.  (Same for the others.)